confdb/confdb.go

```diff
@@ -1431,16 +1587,74 @@ func (p placeholder) String() string {
 	return "{" + string(p) + "}"
 }
 
+type matchedPlaceholders struct {
+	index map[string]string
+	key   map[string]string
+}
+
+// indexPlaceholder represents a subkey of a name/path (e.g., "[{n}]") that can
+// match an index value and map it from the input name to the path.
+type indexPlaceholder string
+
+// match checks if the subkey can be used to index a list. If so, it adds a
+// mapping from this placeholder key to the supplied name subkey and returns true.
+func (p indexPlaceholder) match(subkey string, matched *matchedPlaceholders) bool {
+	if len(subkey) < 3 || !(subkey[0] == '[' && subkey[len(subkey)-1] == ']') {
+		return false
+	}
+
+	if _, err := strconv.Atoi(subkey[1 : len(subkey)-1]); err != nil {
+		// subkey can't be used as a index placeholder value
+		return false
+	}
```

Comment:
reuse regex?

Reply:
used the helpers as mentioned in another comment